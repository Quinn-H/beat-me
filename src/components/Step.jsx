import React from 'react'

const Step = (props) => {
  const step = props.step
  const isSet = step.isSet
  const isCurrent = step.isCurrent
  const onClick = step.onClick
  const buttonType = isCurrent ? 'warning' : 'primary'
  const className = `btn ${isSet ? 'btn-' : 'btn-outline-'}${buttonType}`
  return (
    <td>
      <button
        type="button"
        className={className}
        onClick={onClick}>
      </button>
    </td>
  )
}

export default Step
