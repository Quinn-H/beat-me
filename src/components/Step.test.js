/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Step from './Step'

test('Step renders correctly', () => {
  const onClick = function () {}
  const isSet = false
  const isCurrent = false
  const step = {onClick, isSet, isCurrent}
  const tree = renderer.create(
    <Step step={step} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
