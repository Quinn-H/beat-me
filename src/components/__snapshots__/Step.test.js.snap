exports[`test Step renders correctly 1`] = `
<td>
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
    type="button" />
</td>
`;
